server/service/handler.go

```diff
@@ -503,7 +503,7 @@ func MakeHandler(svc fleet.Service, config config.FleetConfig, logger kitlog.Log
 			setRequestsContexts(svc),
 		),
 		kithttp.ServerErrorHandler(&errorHandler{logger}),
-		kithttp.ServerErrorEncoder(encodeError),
+		kithttp.ServerErrorEncoder(errorEncoder(logger)),
 		kithttp.ServerAfter(
 			kithttp.SetContentType("application/json; charset=utf-8"),
 			logRequestEnd(logger),
```

Comment:
Hm, I'm not sure I understand what's happening. This `logRequestEnd` should've logged the error regardless of the log level.